test/integration/buildstrategy_to_taskruns_test.go

```diff
@@ -390,6 +390,46 @@ var _ = Describe("Integration tests BuildStrategies and TaskRuns", func() {
 			Expect(br.Status.GetCondition(v1alpha1.Succeeded).GetMessage()).To(Equal("parameters without a value definition: sleep-time"))
 
 		})
-
+		Context("when a taskrun fails with an error result", func() {
+			It("surfaces the result to the buildrun", func() {
+				// Create a BuildStrategy that guarantees a failure 
+				cbsObject, err := tb.Catalog.LoadBuildStrategyFromBytes(
+					[]byte(test.BuildStrategyWithErrorResult),
+				)
+				Expect(err).To(BeNil())
+
+				err = tb.CreateBuildStrategy(cbsObject)
+				Expect(err).To(BeNil())
+
+				buildObject, err := tb.Catalog.LoadBuildWithNameAndStrategy(
+					BUILD+tb.Namespace,
+					cbsObject.Name,
+					[]byte(test.BuildBSMinimal),
+				)
+				Expect(err).To(BeNil())
+
+				Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+				_, err = tb.GetBuildTillValidation(buildObject.Name)
+				Expect(err).To(BeNil())
+
+				buildRunObject, err := tb.Catalog.LoadBRWithNameAndRef(
+					BUILDRUN+tb.Namespace,
+					BUILD+tb.Namespace,
+					[]byte(test.MinimalBuildRun),
+				)
+				Expect(err).To(BeNil())
+
+				Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+
+				buildRun, err := tb.GetBRTillCompletion(buildRunObject.Name)
+				Expect(err).To(BeNil())
+
+				Expect(buildRun.Status.FailureDetails.Location.Container).To(Equal("step-fail-with-error-result"))
+				Expect(buildRun.Status.FailureDetails).ToNot(BeNil())
```

Comment:
Better to add `FailureDetails` nil check before accessing something from `FailureDetails`. In short please move one line above so that we will access the `FailureDetails.Location` only after the `nil` check of `FailureDetails` field.